eos-live-boot-generator

```diff
@@ -11,14 +11,68 @@ is_live()
     grep -q "\<endless.live_boot\>" "${proc_cmdline:-/proc/cmdline}"
 }
 
-dest_dir="${1:?normal-dir argument missing}"
+if ! is_live; then
+    exit 0
+fi
 
+dest_dir="${1:?normal-dir argument missing}"
 system_dir=/usr/lib/systemd/system
 local_fs_target_wants="$dest_dir/local-fs.target.wants"
-if is_live; then
-    mkdir -p "$local_fs_target_wants"
-    ln -sf "$system_dir/systemd-udev-settle.service" "$local_fs_target_wants/"
 
-    # Don't try to remount the (read-only) root filesystem read-write
-    ln -sf /dev/null "$dest_dir/systemd-remount-fs.service"
-fi
+mkdir -p "$local_fs_target_wants"
+ln -sf "$system_dir/systemd-udev-settle.service" "$local_fs_target_wants/"
+
+# Don't try to remount the (read-only) root filesystem read-write
+ln -sf /dev/null "$dest_dir/systemd-remount-fs.service"
+
+cat << MOUNT_EOSLIVE > "$dest_dir/run-media-eoslive.mount"
+[Unit]
+Description=Endless Live USB Storage
+ConditionKernelCommandLine=endless.image.device
+
+[Mount]
+What=/dev/mapper/endless-image-device
+Where=/run/media/eoslive
+MOUNT_EOSLIVE
+
+cat << MOUNT_KOLIBRIHOME > "$dest_dir/run-kolibrihome.mount"
+[Unit]
+Description=Kolibri data from Endless Live USB Storage
+ConditionKernelCommandLine=endless.image.device
+ConditionDirectoryNotEmpty=/run/media/eoslive/.kolibri
+
+[Mount]
+What=/run/media/eoslive/.kolibri
+Where=/run/kolibrihome
+Options=bind
+MOUNT_KOLIBRIHOME
+
+cat << SETUP_KOLIBRIHOME > "$dest_dir/eoslive-kolibrihome-setup.service"
+[Unit]
+Description=Setup Kolibri to use data from Endless Live USB Storage
+Requires=run-kolibrihome.mount
+After=systemd-tmpfiles-setup.service
+ConditionKernelCommandLine=endless.image.device
+ConditionDirectoryNotEmpty=/run/kolibrihome
+
+[Service]
+Type=oneshot
+ExecStart=/bin/bash -c 'echo -e "[Environment]\nKOLIBRI_HOME=/run/kolibrihome" > /run/flatpak/overrides/org.learningequality.Kolibri'
+
+[Install]
+WantedBy=multi-user.target
```

Comment:
Is just writing the service to `$dest_dir` sufficient to cause it to be activated? Does `systemctl preset` or `preset-all` (or the moral equivalent) get run after the generators run? Naïvely I would expect to need to put it into `multi-user.target.wants/` but I am always surprised by these things…

Reply:
Yep, you are right -- we can't call `systemctl` from generators since d-bus is not around yet, but we should be manually enabling these by creating syslinks in the right place, i.e., `/run/systemd/generator/multi-user.target.wants`.

I'm fixing this atm.